internal/matchers/expectation.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019, OpenTelemetry Authors
```

Comment:
I imagine you'd secretly like to use another testing package, but have written this to avoid it?

Reply:
FWIW: I wouldn't accept this w/o a good explanation. `interface{}` means  these may not always work the way you think they will. For instance see the hoops that testify's Assert lib goes through .